spec/services/pagination_service_spec.rb

```diff
@@ -0,0 +1,689 @@
+            expect(response['links']["self"]).to eq first_page
+            expect(response['links']["first"]).to eq first_page
+            expect(response['links']["prev"]).to be_nil
+            expect(response['links']["next"]).to eq second_page
+            expect(response['links']["last"]).to eq last_page
+          end
+
+          it 'does not include errors' do
+            expect(response.key?('errors')).to eq false
+          end
+        end
+
+        context 'and page_offset is start of second page' do
+          let(:page_offset) { "11" }
+
+          it 'returns json api response second page of results' do
+            expect(response["data"]).to match_array(results[10..19])
+          end
+
+          it "sets meta['page'] stats" do
+            expect(response["meta"]["page"]["page_offset"]).to eq "11"
+            expect(response["meta"]["page"]["page_limit"]).to eq "10"
+            expect(response["meta"]["page"]["actual_page_size"]).to eq "10"
+            expect(response["meta"]["page"]["total_num_found"]).to eq "36"
+          end
+
+          it 'sets links with prev and next having values' do
+            expect(response['links']["self"]).to eq second_page
+            expect(response['links']["first"]).to eq first_page
+            expect(response['links']["prev"]).to eq first_page
+            expect(response['links']["next"]).to eq third_page
+            expect(response['links']["last"]).to eq last_page
+          end
+
+          it 'does not include errors' do
+            expect(response.key?('errors')).to eq false
+          end
+        end
+
+        context 'and page_offset is start of third page' do
+          let(:page_offset) { "21" }
+
+          it 'returns json api response third page of results' do
+            expect(response["data"]).to match_array(results[20..29])
+          end
+
+          it "sets meta['page'] stats" do
+            expect(response["meta"]["page"]["page_offset"]).to eq "21"
+            expect(response["meta"]["page"]["page_limit"]).to eq "10"
+            expect(response["meta"]["page"]["actual_page_size"]).to eq "10"
+            expect(response["meta"]["page"]["total_num_found"]).to eq "36"
+          end
+
+          it 'sets links with next set to nil' do
+            expect(response['links']["self"]).to eq third_page
+            expect(response['links']["first"]).to eq first_page
+            expect(response['links']["prev"]).to eq second_page
+            expect(response['links']["next"]).to eq last_page
+            expect(response['links']["last"]).to eq last_page
+          end
+
+          it 'does not include errors' do
+            expect(response.key?('errors')).to eq false
+          end
+        end
+
+        context 'and page_offset is start of fourth page (last page)' do
+          let(:page_offset) { "31" }
+
+          it 'returns json api response third page of results' do
+            expect(response["data"]).to match_array(results[30..39])
```

Comment:
Interesting that this works if results only goes to 35.  Maybe it would still be worth updating to be clear?
```suggestion
            expect(response["data"]).to match_array(results[30..35])
```

Reply:
This is why this worked...
```
>> results[30..39]
[{:id=>"31", :label=>"term 31"}, {:id=>"32", :label=>"term 32"}, {:id=>"33", :label=>"term 33"}, {:id=>"34", :label=>"term 34"}, {:id=>"35", :label=>"term 35"}, {:id=>"36", :label=>"term 36"}]
```

But I definitely agree that it is better to update it to the expected range.